src/tests/ftest/recovery/pool_membership.py

```diff
@@ -0,0 +1,205 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+
+from ClusterShell.NodeSet import NodeSet
+
+from apricot import TestWithServers
+from general_utils import report_errors
+from run_utils import run_remote
+
+
+class PoolMembershipTest(TestWithServers):
+    """Test Pass 2: Pool Membership
+
+    :avocado: recursive
+    """
+
+    def get_rank_to_free(self):
+        """Call dmg storage query usage for all servers and return free space
+        (avail_bytes) for each rank as dictionary.
```

Comment:
Docstring summaries should be one line.  The details about how something is accomplished can be included after the summary.  How does this sound?
```suggestion
        """Get the free space for each engine rank.
        
        Call dmg storage query usage for all servers and return free space
        (avail_bytes) for each rank as dictionary.
```

Note: pydocstyle is useful for catching these issues and can be installed in VSC, but we currently don't require it.

Reply:
Done. I searched `pydocstyle` in VSC, but didn't come up.